includes/collections/class-post-type.php

```diff
@@ -186,4 +189,142 @@ public static function output_collection_meta_data_for_admin_scripts( $current_s
 			);
 		}
 	}
+
+	/**
+	 * Check if a collection title already exists.
+	 *
+	 * @param string $title     The title to check.
+	 * @param int    $exclude_id Post ID to exclude from the check.
+	 * @return bool True if a collection with this title exists, false otherwise.
+	 */
+	public static function title_exists( $title, $exclude_id = 0 ) {
+		$args = [
+			'post_type'      => self::get_post_type(),
+			'post_status'    => [ 'publish', 'private', 'future', 'pending', 'draft' ],
+			'posts_per_page' => -1,
+			'fields'         => 'ids',
```

Comment:
Can we use the title as `s` argument here? This should also allow us to set a reasonable `post_per_page` so this query is less heavy.

Reply:
Ah TIL - thanks @miguelpeixe! I ended up setting the `posts_per_page` to 100 since that matches a couple other uses in the Newspack Plugin -- I'm flip flopping between whether that's still excessive, or a nice buffer "just in case". Let me know if that seems like a lot!